src/moscot/problems/space/_alignment.py

```diff
@@ -128,23 +132,56 @@ def prepare(
         """
         self.spatial_key = spatial_key
         self.batch_key = batch_key
-
+        if policy == "star" and reference is None:
+            raise ValueError("Reference must be provided when `policy='star'`.")
         x = y = {"attr": "obsm", "key": self.spatial_key}
 
-        if normalize_spatial and "x_callback" not in kwargs and "y_callback" not in kwargs:
-            kwargs["x_callback"] = kwargs["y_callback"] = "spatial-norm"
-            kwargs.setdefault("x_callback_kwargs", x)
-            kwargs.setdefault("y_callback_kwargs", y)
-
-        if "spatial-norm" in kwargs.get("x_callback", {}) and "spatial-norm" in kwargs.get("y_callback", {}):
-            x = {}
-            y = {}
-
-        xy, kwargs = handle_joint_attr(joint_attr, kwargs)
-        xy, x, y = handle_cost(xy=xy, x=x, y=y, cost=cost, cost_kwargs=cost_kwargs, **kwargs)  # type: ignore[arg-type]
-
+        if normalize_spatial:
+            if x_callback is None and y_callback is None:
+                x_callback = y_callback = "spatial-norm"
+                if not len(x_callback_kwargs):
+                    x_callback_kwargs = x
+                if not len(y_callback_kwargs):
+                    y_callback_kwargs = y
+            else:
+                raise ValueError("Cannot normalize spatial coordinates when `x_callback` and `y_callback` are set.")
+                warnings.warn(
+                    "Ignoring `normalize_spatial` as `x_callback` and `y_callback` are set.",
+                    UserWarning,
+                    stacklevel=2,  # TODO: check these later
+                )
+        # TODO: what if one is spatial norm?
+        if x_callback == "spatial-norm" and y_callback == "spatial-norm":
+            x = y = {}
+        xy, xy_callback, xy_callback_kwargs = handle_joint_attr(joint_attr, xy_callback, xy_callback_kwargs)
```

Comment:
> `# TODO: what if one is spatial norm?`

it must be in either both or none, as there is no use case for only one, hence they are set both by `normalize_spatial` arg

also this apply only for the alignment, as in the mapping it can be only in one

Reply:
so should I throw an error when only one is given?

Reply:
also for spatiotemporal. 
I think a warning would be good